Makefile

```
@@ -0,0 +1,40 @@
+
+MAKEFILE_ROOT=$(shell pwd)
+
+# install: Ensure that the Argo CD namespace exists, and that CRDs we are using are applied to the current cluster
+install:
+	kubectl create ns argocd || true
+	kubectl apply -f $(MAKEFILE_ROOT)/backend/config/crd/bases
+	kubectl apply -f $(MAKEFILE_ROOT)/backend-shared/config/crd/bases/managed-gitops.redhat.com_operations.yaml
+	kubectl apply -f https://raw.githubusercontent.com/argoproj/argo-cd/release-2.2/manifests/crds/application-crd.yaml
+
+# start: start all the components
+start:
+	~/go/bin/goreman start
+
+# build: build all the components (note: you do not need to do this before running start; 'start' will compile and run)
+build:
+	cd $(MAKEFILE_ROOT)/backend && make build
+	cd $(MAKEFILE_ROOT)/cluster-agent && make build
+
+# test: run tests for each component
+test:
+	cd $(MAKEFILE_ROOT)/backend && make test
+	cd $(MAKEFILE_ROOT)/backend-shared && make test	
+	cd $(MAKEFILE_ROOT)/cluster-agent && make test
+
+# download-deps: Download goreman to ~/go/bin
+download-deps:
```

```suggestion
download-deps: ## Download goreman to ~/go/bin
```

Sugar syntax for `make help` to work.